test/extensions/filters/network/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,194 @@
+  read_filter_callbacks_->connection_.remote_address_ = addr_;
+  read_filter_callbacks_->connection_.local_address_ = addr_;
+  async_request_ = std::make_unique<Grpc::MockAsyncRequest>();
+}
+
+UberFilterFuzzer::UberFilterFuzzer() : time_source_(factory_context_.simulatedTimeSystem()) {
+  fuzzerSetup();
+}
+bool UberFilterFuzzer::invalidInputForFuzzer(const std::string& filter_name,
+                                             Protobuf::Message* config_message) {
+  // System calls such as reading files are prohibited in this fuzzer. Some input that crashes the
+  // mock/fake objects are also prohibited.
+  const std::string name = Extensions::NetworkFilters::Common::FilterNameUtil::canonicalFilterName(
+      std::string(filter_name));
+  if (filter_name == NetworkFilterNames::get().DirectResponse) {
+    envoy::extensions::filters::network::direct_response::v3::Config& config =
+        dynamic_cast<envoy::extensions::filters::network::direct_response::v3::Config&>(
+            *config_message);
+    if (config.response().specifier_case() ==
+        envoy::config::core::v3::DataSource::SpecifierCase::kFilename) {
+      return true;
+    }
+  } else if (filter_name == NetworkFilterNames::get().LocalRateLimit) {
+    envoy::extensions::filters::network::local_ratelimit::v3::LocalRateLimit& config =
+        dynamic_cast<envoy::extensions::filters::network::local_ratelimit::v3::LocalRateLimit&>(
+            *config_message);
+    if (config.token_bucket().fill_interval().seconds() > seconds_in_one_day_) {
+      // Too large fill_interval may cause "c++/v1/chrono" overflow when simulated_time_system_ is
+      // converting it to a smaller unit. Constraining fill_interval to no greater than one day is
+      // reasonable.
+      return true;
+    }
+  }
+  return false;
+}
+
+void UberFilterFuzzer::fuzz(
+    const envoy::config::listener::v3::Filter& proto_config,
+    const Protobuf::RepeatedPtrField<::test::extensions::filters::network::Action>& actions) {
+  try {
+    // Try to create the filter callback(cb_). Exit early if the config is invalid or violates PGV
+    // constraints.
+    const std::string& filter_name = proto_config.name();
+    ENVOY_LOG_MISC(info, "filter name {}", filter_name);
+    auto& factory = Config::Utility::getAndCheckFactoryByName<
+        Server::Configuration::NamedNetworkFilterConfigFactory>(filter_name);
+    ProtobufTypes::MessagePtr message = Config::Utility::translateToFactoryConfig(
+        proto_config, factory_context_.messageValidationVisitor(), factory);
+    if (invalidInputForFuzzer(filter_name, message.get())) {
+      // Make sure no invalid system calls are executed in fuzzer.
+      return;
+    }
+    ENVOY_LOG_MISC(info, "Config content after decoded: {}", message->DebugString());
+    cb_ = factory.createFilterFactoryFromProto(*message, factory_context_);
+    perFilterSetup(proto_config.name());
+    // Add filter to connection_.
+    cb_(read_filter_callbacks_->connection_);
```

Comment:
Does perFilterSetup and the following line to create and set `read_filter_` need to occur in this try/catch? I don't think they should be throwing exceptions

Reply:
> Does perFilterSetup and the following line to create and set `read_filter_` need to occur in this try/catch? I don't think they should be throwing exceptions

Yes they shouldn't throw exceptions.(unless they do system calls like reading a file that doesn't exist)